### PR TITLE
Refactors /turf/var/intact, cleans up some hide issues

### LINKED ISSUE #8766: Plumbing devices' pipes have strange hiding behaviour
## Reproduction:
Almost certainly a result of [8266](https://github.com/BeeStation/BeeStation-Hornet/pull/8266), plumbing devices now default to having their inputs hidden, making it very tedious to manipulate them without prying the floors out everywhere.
image

Furthermore, the behaviour involving them is... odd. Prying the tile they're sitting on will reveal the pipe even if the device is moved...but placing it back down will hide the pipe again even after the device was moved off it so you can end up with a scuffed situation like this.
image

Even if some of the behaviour is intended, this is confusing and unintuitive at the very least.

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/52158
- https://github.com/tgstation/tgstation/pull/62331
- https://github.com/tgstation/tgstation/pull/64346

## About The Pull Request
This pr further ports underfloor fixes, preparing us for glass floor tiles and tg's layered plumbing features.

Refactors the intact var into three different defines. 

Fixes manifolds hiding under floors

TODO
- [x] Base port done
- [x] expected plumbing behavior
- [x] expected wiring behavior
- [x] expected RPD behavior
- [x] ready to review

## Why It's Good For The Game
intact is a very annoying var to work around with its double usages

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Pipes and wires hide and interact correctly
https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/c21da7ed-3247-4b7f-821f-74b88e244bdf

Satchels hide under the correct areas
hiddensatchel

plumbing doesnt erroneously hide

image

</details>

## Changelog
:cl: RKz, esaine, LemonInTheDark, magatsuchi
refactor: Reworked /turf/var/intact into variables which don't conflate underfloor visibility, underfloor interactability, and overfloor presence
fix: things that are supposed to affect things under tiles do, things that arent wont
fix: 'hide equals' issues
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
merge #8866 first

**Comment by Tsar-Salat** [comment]
wow, I forgor about this pr.

Ill get to it soon

**Comment by Tsar-Salat** [comment]
> Ill get to it soon

Finally got to it.

lazy bastard

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -678,7 +678,7 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list (new/datum/stack_recipe("cable restrai
 			if (showerror)
 				to_chat(user, "<span class='warning'>You can only lay cables on catwalks and plating!</span>")
 			return
-		if(U.intact)						//can't place a cable if it's a plating with a tile on it
+		if(U.underfloor_accessibility < UNDERFLOOR_HIDDEN)//can't place a cable if it's a plating with a tile on it
```
Comment (itsmeow) [inline code/modules/power/cable.dm]: isn't nothing less than this?

Comment (Tsar-Salat) [reply code/modules/power/cable.dm]: Theres already a `T.underfloor_accessibility < UNDERFLOOR_INTERACTABLE` check above, so this is just duplicate code, even on our current master. If the underfloor is not accessible(which is connected to t-ray scanning and whatnot) the sanity check will already hit it.

Good catch. TG cleaned up all this code in smartcables, its a 13 year old check that got duplicated at some point.

tl;dr 13 yr old duplicate code fuckin aaaaaaaaaaa

image

File: code/modules/power/cable.dm [hunk 2]
```diff
@@ -678,9 +678,6 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list (new/datum/stack_recipe("cable restrai
 			if (showerror)
 				to_chat(user, "<span class='warning'>You can only lay cables on catwalks and plating!</span>")
 			return
-		if(U.intact)						//can't place a cable if it's a plating with a tile on it
-			to_chat(user, "<span class='warning'>You can't lay cable there unless the floor tiles are removed!</span>")
-			return
 		else
 			// cable is pointing at us, we're standing on an open tile
```
Comment (itsmeow) [inline code/modules/power/cable.dm:688]: The else block here is now useless since the if early returns.

File: code/modules/procedural_mapping/mapGenerators/repair.dm [hunk 3]
```diff
@@ -56,7 +56,7 @@
 	// This is a manual verb so performance is not a priority.
 	for(var/turf/updated_turf as anything in block(locate(bounds[MAP_MINX], bounds[MAP_MINY], SSmapping.station_start), locate(bounds[MAP_MAXX], bounds[MAP_MAXY], z_offset - 1)))
 		for(var/atom/contained_atom as anything in updated_turf)
-			if(isobj(contained_atom) && !istype(updated_turf, /turf/open/space) && updated_turf.intact)
+			if(isobj(contained_atom) && !istype(updated_turf, /turf/open/space) && updated_turf.underfloor_accessibility < UNDERFLOOR_INTERACTABLE)
```
Comment (itsmeow) [inline code/modules/procedural_mapping/mapGenerators/repair.dm]: This should be UNDERFLOOR_VISIBLE, right? This is really supposed to recreat the behavior of levelupdate()

**Comment by Tsar-Salat** [comment]
im taking a break, Semester begins soon and this is affecting my mental health

**Comment by Tsar-Salat** [comment]
woaw, how the hell is this old ass branch not merge conflicting. This is amazing

File: code/modules/procedural_mapping/mapGenerators/repair.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/modules/procedural_mapping/mapGenerators/repair.dm]: changed in 5a81dc1

File: code/modules/power/cable.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/power/cable.dm:684]: 5a81dc1

Comment (Tsar-Salat) [reply code/modules/power/cable.dm:688]: Yes you are correct.

I added it back.

2b226e1

**Review by itsmeow — APPROVED** [review]
